### PR TITLE
Refactoring PR + Random Walker Implementation and Hyperparameter Optimization

### PR DESCRIPTION
**Pull request by debapratimj**

### CODE REVIEW

File: abm/exploration_strategy.py
Comment (vagechirkov) [inline abm/exploration_strategy.py:55]: TODO: add the parameter explanations

File: abm/tests/test_exploration_strategy.py
Comment (vagechirkov) [inline abm/tests/test_exploration_strategy.py]: Please change the testing framework to `pytest`

**Review by vagechirkov — CHANGES_REQUESTED** [review]
Needs to be done before merging:
- Add parameter descriptions and references to the paper in docstring
- Replace `unittest` with the `pytest` testing framework
- Restore `utils` tests

File: abm/exploration_strategy.py
Comment (debapratimj) [reply abm/exploration_strategy.py:55]: added docstring

File: abm/tests/test_exploration_strategy.py
Comment (debapratimj) [reply abm/tests/test_exploration_strategy.py]: shifted to pytest

**Comment by debapratimj** [comment]
Waiting for merge with owner with write access

File: abm/agent.py
Comment (vagechirkov) [inline abm/agent.py]: I don't think we should override model parameter here!

File: abm/agent.py
Comment (vagechirkov) [inline abm/agent.py:132]: I am wondering, should we not convert the destination back from ij to xy format here? This could be a subtle bug!

File: abm/exploration_strategy.py
Comment (vagechirkov) [inline abm/exploration_strategy.py]: Order is important!

**Comment by vagechirkov** [comment]
LGTM, thanks @debapratimj!

**Pull request merged**
